src/main/java/com/serhiiklymchuk/gmailpet/controller/UserController.java

```diff
@@ -0,0 +1,44 @@
+package com.serhiiklymchuk.gmailpet.controller;
+
+import com.serhiiklymchuk.gmailpet.dto.UserDto;
+import com.serhiiklymchuk.gmailpet.service.impl.UserServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import javax.validation.Valid;
+
+@Controller
+@RequestMapping("/user")
+public class UserController {
+    @Autowired
+    UserServiceImpl userService;
+
+    @GetMapping("/home")
+    public String getHome(){
+        return "user/home";
+    }
+
+    @GetMapping("/register")
+    public String getRegistrationPage(){
+        return "user/register";
+    }
+
+    @PostMapping("/register")
+    public String createUser(@Valid UserDto userDto, BindingResult bindingResult, RedirectAttributes attr, Model model){
+        if(bindingResult.hasErrors()){
+            model.addAttribute("errors", bindingResult.getAllErrors());
+            return "user/register";
```

Comment:
It's actually, acceptable, to render the page on POST request, if the request didn't pass validation.
I do not prefer this style, but it can be. I've seen this couple of times in Ruby on Rails applications.